### PR TITLE
add otel provider in demo profile

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

add `otel` provider in demo profile, make a doc test easier.
also this PR is will be used by https://github.com/istio/istio.io/pull/10977
@douglas-reid PTAL.

### CODE REVIEW

File: samples/open-telemetry/README.md [hunk 1]
```diff
@@ -31,43 +31,27 @@ service:
...
 ```
 
-## Modiy istio configmap
+## Update istio configmap
 
-Run the following script to edit the istio MeshConfig, update the YAML files in one step.
+Run the following script to update the `istio` with demo profile:
+
+```bash
+istioctl install -f manifests/profiles/demo.yaml -y
```
Comment (howardjohn) [inline samples/open-telemetry/README.md:39]:
```suggestion
istioctl install --profile demo -y
```

**Review by howardjohn — APPROVED** [review]

**Comment by zirain** [comment]
@craigbox can you take a look at this?

this PR will push forward https://github.com/istio/istio.io/pull/10977

**Review by kebe7jun — APPROVED** [review]

**Comment by craigbox** [comment]
If you want your new page in the current version of the docs, will this need to be backported to a 1.13 release?  (Not sure how they are packaged)

**Pull request merged**